js.go

```diff
@@ -2243,7 +2247,12 @@ func (sub *Subscription) Fetch(batch int, opts ...PullOpt) ([]*Msg, error) {
 			if err == nil {
 				var usrMsg bool
 
-				usrMsg, err = checkMsg(msg, true)
+				// If we already got at least a message but we are prompted
+				// with a 408 status, then unblock the request and return
+				// the messages that have arrived already.
+				cancelWhen408 := len(msgs) >= 1
+
+				usrMsg, err = checkMsg(msg, true, cancelWhen408)
```

Comment:
If you do without cancelWhen408, just check for err == ErrTimeout at the end if this if/else by adding:
```
..
} else if err == ErrTimeout && len(msgs) == 0 {
    err = nil
}
```
Keep in mind that even if we get an error in this function, at the end we will return no error if `len(msgs)` is > 0 (see at the end).

Reply:
I didn't get this change completely but sounds interesting, could you share me the whole snippet of where to apply the change?

Reply:
Remove the cancelWhen408 variable definition above and restore checkMsg() (remove boolean), but set `err = ErrTimeout` when processing 408 header.
Then here the code would be:
```
usrMsg, err = checkMsg(msg, true)
if err == nil && usrMsg {
    msgs = append(msgs, msg)
} else if noWait && (err == errNoMessages) && len(msgs) == 0 {
    // If we have a 404 for our "no_wait" request and have
    // not collected any message, then resend request to
    // wait this time.
    noWait = false
    err = sendReq()
} else if err == ErrTimeout && len(msgs) == 0 {
    // If we get a 408, we will bail if we already collected some
    // messages, otherwise ignore and go back calling nextMsg
    err = nil
}
```

Reply:
Thanks! I updated now.